DashAI/front/src/components/experiments/PrepareDatasetStep.jsx

```diff
@@ -0,0 +1,398 @@
+ * @param {function} setNextEnabled function to enable or disable the "Next" button in the modal
+ */
+function PrepareDatasetStep({ newExp, setNewExp, setNextEnabled }) {
+  // TODO: column and row numbers should be minor to the maximum on the dataset
+  const totalColumns = 100;
+  const totalRows = 2000;
+  // columns numbers state
+  const [inputColumns, setInputColumns] = useState([]);
+  const [outputColumns, setOutputColumns] = useState([]);
+  const [columnsReady, setColumnsReady] = useState(false);
+
+  // handle columns numbers change state
+  const [parseInputColumnsError, setParseInputColumnsError] = useState(false);
+  const [parseOutputColumnsError, setParseOutputColumnsError] = useState(false);
+  const [parseInputColumnsErrorText, setParseInputColumnsErrorText] =
+    useState("");
+  const [parseOutputColumnsErrorText, setParseOutputColumnsErrorText] =
+    useState("");
+  const rangeRegex = /^(\d+)(-(\d+))*(,(\d+)(-(\d+))*)*$/;
+
+  // rows numbers state
+  const defaultParitionsIndex = {
+    training: [],
+    validation: [],
+    testing: [],
+  };
+  const defaultPartitionsPercentage = {
+    training: 0,
+    validation: 0,
+    testing: 0,
+  };
+  const [rowsPartitionsIndex, setRowsPartitionsIndex] = useState(
+    defaultParitionsIndex,
+  );
+  const [rowsPartitionsPercentage, setRowsPartitionsPercentage] = useState(
+    defaultPartitionsPercentage,
+  );
+  const [splitsReady, setSplitsReady] = useState(false);
+
+  // handle rows numbers change state
+  const [rowsPreference, setRowsPreference] = useState("random");
+  const [rowsPartitionsError, setRowsPartitionsError] = useState(false);
+  const [rowsPartitionsErrorText, setRowsPartitionsErrorText] = useState("");
+
+  const parseRangeToIndex = (value, total) => {
+    const numbersArray = [];
+    if (!rangeRegex.test(value)) {
+      throw new Error(
+        "Ranges and indexes must match the format on the example above",
+      );
+    }
+    const ranges = value.split(",");
+    ranges.forEach((range) => {
+      const [min, max] = range.split("-");
+      if (!range.includes("-") && parseInt(range) <= total) {
+        numbersArray.push(parseInt(range));
+      } else if (
+        (!range.includes("-") && parseInt(range) > total) ||
+        parseInt(max) > total
+      ) {
+        throw new Error(
+          "The indexes should be minor than the total of columns or rows",
+        );
+      } else if (parseInt(min) > parseInt(max)) {
+        throw new Error(
+          "The second number of a range must be greater than the first",
+        );
+      } else {
+        for (let i = parseInt(min); i <= parseInt(max); i++) {
+          numbersArray.push(i);
+        }
+      }
+    });
+    return numbersArray;
+  };
+
+  const checkSplit = (training, validation, testing) => {
+    if (parseInt(training) + parseInt(validation) + parseInt(testing) === 100) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
It can be shortened a little without losing readability.
```suggestion
    return parseInt(training) + parseInt(validation) + parseInt(testing) === 100
```